shark/src/main/java/shark/internal/PathFinder.kt

```diff
@@ -462,6 +473,51 @@ internal class PathFinder(
     }
   }
 
+  private fun HeapInstance.readAllNonNullFieldsOfReferenceType(): MutableList<HeapField> {
+    val hprofGraph = graph // DO NOT REMOVE! Getter performs typecast which is expensive!
```

Comment:
- always put line related comments on the line before, never the end of a line, as those will eventually wrap and comment *after* the code.
- Remove `DO NOT REMOVE!`, avoid scary language that stops people from thinking. Explain instead:

```
// Assigning to local variable to avoid repeated lookup and cast (HeapInstance.graph casts HeapInstance.hprofGraph to HeapGraph in is getter)
val hprofGraph = graph
```

Reply:
note from the Jake: 

> D8 in release mode and R8 have full type awareness and will remove redundant checkcast instructions